tests/RabbitMQJobTest.php

```diff
@@ -0,0 +1,175 @@
+            ->shouldReceive('getRoutingKey')
+            ->andReturn('#.test_v1.handle_test_1');
+
+        $this->amqpMessageMock
+            ->shouldReceive('getBody')
+            ->andReturn(json_encode([]));
+
+        $rabbitMqJob = new RabbitMQJob(
+            $this->containerMock,
+            $this->rabbitmqMock,
+            $this->amqpMessageMock,
+            'test-connection',
+            'test-queue'
+        );
+
+        $this->assertEquals(TestHandlerOne::class, $rabbitMqJob->payload()['job']);
+    }
+
+    /** @test */
+    public function whenGetPayloadItShouldReturnTheCorrectJob()
+    {
+        $this->amqpMessageMock
+            ->shouldReceive('getRoutingKey')
+            ->andReturn('#.test_v1.handle_test_4');
+
+        $this->amqpMessageMock
+            ->shouldReceive('getBody')
+            ->andReturn(json_encode([]));
+
+        $rabbitMqJob = new RabbitMQJob(
+            $this->containerMock,
+            $this->rabbitmqMock,
+            $this->amqpMessageMock,
+            'test-connection',
+            'test-queue'
+        );
+
+        $this->assertEquals(TestHandlerTwo::class, $rabbitMqJob->payload()['job']);
+    }
+
+    /** @test */
+    public function whenGetPayloadItShouldReturnTheExpectedPayload()
+    {
+        $this->amqpMessageMock
+            ->shouldReceive('getRoutingKey')
+            ->andReturn('#.test_v1.handle_test_1');
+
+        $this->amqpMessageMock
+            ->shouldReceive('getBody')
+            ->andReturn(json_encode(['id' => 'test_1', 'version' => 'v1']));
+
+        $rabbitMqJob = new RabbitMQJob(
+            $this->containerMock,
+            $this->rabbitmqMock,
+            $this->amqpMessageMock,
+            'test-connection',
+            'test-queue'
+        );
+
+        $this->assertEquals(
+            [
+                'job'  => TestHandlerOne::class,
+                'data' => [
+                    'id'      => 'test_1',
+                    'version' => 'v1',
+                ],
+            ],
+            $rabbitMqJob->payload()
+        );
+    }
+
+    /** @test */
+    public function whenFireItShouldDispatchAllHandlers()
+    {
+        $this->amqpMessageMock
+            ->shouldReceive('getRoutingKey')
+            ->andReturn('#.test_v1.handle_test_1');
+
+        $this->amqpMessageMock
+            ->shouldReceive('getBody')
+            ->andReturn(json_encode(['id' => 'test_1', 'version' => 'v1']));
+
+        $this->rabbitmqMock
+            ->shouldReceive('ack')->once();
+
+        $rabbitMqJob = new RabbitMQJob(
+            $this->containerMock,
+            $this->rabbitmqMock,
+            $this->amqpMessageMock,
+            'test-connection',
+            'test-queue'
+        );
+
+        $rabbitMqJob->fire();
+
+        Queue::assertPushed(TestHandlerOne::class);
+        Queue::assertPushed(TestHandlerTwo::class);
+    }
+
+    /** @test */
+    public function whenFireWithoutAssignedHandlerItShouldNotDispatchAnyHandler()
```

Comment:
I think if there is no message handler configured should throw an exception in order to get knowledge about missing parametrization. What do you think?

Reply:
We do not want to throw exception because the queue would be stuck, and it is not desirable, for each message not configured (which should never happen), a message is logged and the message discarded from the queue.